tests/src/test/java/alluxio/testutils/PersistenceTestUtils.java

```diff
@@ -92,9 +94,14 @@ public static void pauseScheduler(LocalAlluxioClusterResource resource) {
    */
   public static void resumeScheduler(LocalAlluxioClusterResource resource) {
     FileSystemMaster nestedMaster = getFileSystemMaster(resource);
-    BlackHoleMap<Long, ExponentialTimer> persistRequests =
-        Whitebox.getInternalState(nestedMaster, "mPersistRequests");
-    Whitebox.setInternalState(nestedMaster, "mPersistRequests", persistRequests.getInnerMap());
+    try {
+      BlackHoleMap<Long, ExponentialTimer> persistRequests =
+          Whitebox.getInternalState(nestedMaster, "mPersistRequests");
+      Whitebox.setInternalState(nestedMaster, "mPersistRequests", persistRequests.getInnerMap());
+    } catch (ClassCastException e) {
+      // This will occur is resume is called without first being paused
+      LOG.warn("Call to resumeScheduler failed: ", e);
```

Comment:
If this happens, does this mean the test was written incorrectly? Should we just throw a real exception to kill the test?

Reply:
Not necessarily - if a test errors between pausing and resuming the scheduler, the test may not necessarily be written incorrectly.

If an Alluxio code changes causes an exception to be thrown in between pausing and resuming it might be possible for pause/resume to be called twice depending on how the next test in the suite is run.